influxql/parser.go

```diff
@@ -1435,8 +1435,8 @@ func (p *Parser) parseOptionalTokenAndInt(t Token) (int, error) {
 	// Parse number.
 	n, _ := strconv.ParseInt(lit, 10, 64)
 
-	if n < 1 {
-		msg := fmt.Sprintf("%s must be > 0", t.String())
+	if n < 0 {
```

Comment:
That was a deliberate design we came to a while back (search Slack).

If my recollection is correct we decided that `LIMIT 0` didn't make much sense, since it should be interpreted literally as "give me no records". SQLite, MySQL (and I believe Postgres) all return 0 records if you ask for `LIMIT 0`. Since there was still arguments about this we simply prohibited `LIMIT 0` in code. Obviously `OFFSET 0` is a different issue, but `OFFSET` did not exist at the time of the discussions.